backend/api/publishing.py

```
@@ -263,10 +650,51 @@ async def save_scenario_draft(
                     # First, remove existing relationships for this scene
                     db.execute(scene_personas.delete().where(scene_personas.c.scene_id == existing_scene.id))
                     
+                    # Helper function to check if persona is the main character (student role)
+                    def is_main_character(persona_name, student_role):
```

Ruff (0.14.1)

Missing return type annotation for private function `is_main_character`

(ANN202)